joplin/js/CreateContentModal/index.js

```diff
@@ -0,0 +1,133 @@
+    this.setState({
+      activeStep: this.state.activeStep + 1
+    });
+  }
+
+  decrementActiveStep = () => {
+    this.setState({
+      activeStep: this.state.activeStep - 1
+    });
+  }
+
+  handleNextButton = (e) => {
+    // Validate title max length
+    if (this.state.titleCharacterCount > MAX_TITLE_LENGTH) return false;
+
+    // Skip Topic Select Step for creating a Department
+    if (this.state.type === 'department' && this.state.activeStep === 1) {
+      this.redirectToEditPage();
+      return true;
+    }
+
+    if (this.state.activeStep === 2) {
+      this.redirectToEditPage();
+      return true;
+    }
+
+    this.incrementActiveStep();
+  }
+
+  handleBackButton = (e) => {
+    this.decrementActiveStep();
+  }
+
+  handleTypeSelect = (dataObj, e) => {
+    this.setState({
+      type: dataObj.type,
+      redirectUrl: dataObj.redirectUrl,
+    });
+    this.incrementActiveStep();
+  }
+
+  handleTitleInputChange = (e) => {
+    this.setState({
+      title: e.target.value,
+      titleCharacterCount: e.target.value.length,
+    });
+  }
+
+  handleTopicSelect = (id) => {
+    this.setState({ topic: id });
+  }
+
+  redirectToEditPage = () => {
+    this.writeToLocalStorage();
+    window.location.href = this.state.redirectUrl;
+  }
+
+  writeToLocalStorage = () => {
+    localStorage.wagtailCreateModal = JSON.stringify(this.state);
+  }
+
+
+  render() {
+    return (
+      <div className="modal fade" id="createNewContentModal" tabIndex="-1" role="dialog" aria-labelledby="createNewContentModalLabel">
+        <div className="CreateContentModal__wrapper">
+          <div className="modal-dialog" role="document">
+            <div className="modal-content CreateContentModal">
+              <div className="modal-body">
+                { this.state.activeStep === 0 &&
+                  <ChooseTypeStep handleTypeSelect={this.handleTypeSelect}/>
+                }
+                { this.state.activeStep === 1 &&
+                  <ChooseTitleStep
+                    pageType={this.state.type}
+                    title={this.state.title}
+                    handleTitleInputChange={this.handleTitleInputChange}
+                    characterCount={this.state.titleCharacterCount}
+                    maxCharacterCount={MAX_TITLE_LENGTH}
+                  />
+                }
+                { this.state.activeStep === 2 &&
+                  <ChooseTopicStep
+                    topic={this.state.topic}
+                    handleTopicSelect={this.handleTopicSelect}
+                    themeTopicTree={THEME_TOPIC_TREE}
+                  />
+                }
+                <ButtonBar
+                  handleBackButton={this.handleBackButton}
+                  handleNextButton={this.handleNextButton}
+                  activeStep={this.state.activeStep}
+                />
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+ReactDOM.render(
```

Comment:
Do we want the ReactDOM.render here? I kind of like the pattern set by create-react-app where you have App.js with your app and then index.js that just imports it and renders it. Not that we need to stick to that naming convention it's just nice to separate the component from the rendering.

Reply:
I don't have a strong opinion about this. Having an extra file that just renders it doesn't seem useful to me but if we wanted to add fancy things like hot module reloading, I think it would become necessary.